build_wheels.py

```diff
@@ -172,28 +177,239 @@ def assemble_requirements(idf_branches: List[str], idf_constraints: List[str]) -
         requirements_txt += _download_branch_requirements(branch, idf_requirements_json)
         requirements_txt += _download_branch_constraints(constraint_file_url, branch, idf_constraints[i])
 
+    if make_txt_file:
+        # TXT file from all downloaded requirements and constraints files
+        # useful for debugging or to see the comments for requirements
+        with open('requirements.txt', 'w') as f:
+            for line in requirements_txt:
+                f.write(line)
+                f.write('\n')
+
     return _add_into_requirements(requirements_txt)
 
 
-def print_requirements(requirements_set: set[Union[Requirement, str]]):
-    """Prints assembled list of requirements"""
-    print('\n---------- REQUIREMENTS ----------')
-    for req in requirements_set:
-        print(req)
+# --- exclude_list and include_list ---
+def _change_specifier_logic(specifier: str) -> str:
+    """Change specifier logic to opposite
+        - e.g. "<3" will be ">3"
+    """
+    new_spec = specifier
+    replacements = {'<': '>',
```

Comment:
You don't use lookup so hashing is an unused overhead. Better use a data types which is better for iteration.

```python
(('<', '>',), ('>', '<'))
```